lighting/raymarch/softShadow.glsl

```diff
@@ -33,28 +32,25 @@ examples:
 #ifndef FNC_RAYMARCH_SOFTSHADOW
 #define FNC_RAYMARCH_SOFTSHADOW
 
-float raymarchSoftShadow(vec3 ro, vec3 rd, in float mint, in float maxt, float w) {
+float raymarchSoftShadow(vec3 ro, vec3 rd) {
+    const float mint = RAYMARCH_SHADOW_MIN_DIST;
+    const float maxt = RAYMARCH_SHADOW_MAX_DIST;
+    const float w = RAYMARCH_SHADOW_SOLID_ANGLE;
+
     float res = 1.0;
     float t = mint;
-    #if defined(PLATFORM_WEBGL)
     for (int i = 0; i < RAYMARCHSOFTSHADOW_ITERATIONS; i++) {
         if (t >= maxt)
             break;
-    #else
-    for (int i = 0; i < RAYMARCHSOFTSHADOW_ITERATIONS && t < maxt; i++) {
```

Comment:
I feel that this is an optimization for non `PLATFORM_WEBGL` I know that makes is harder to read but probably more performant?

Reply:
I'm honestly not a fan of these micro-optimisations @patriciogonzalezvivo we might win a micro-second here (not even sure about that, we'd have to profile it). But we all lose several seconds everytime we have to read this.

I think this is part of a more general conversation about performance vs. readability. I see it this way:
* Lygia is not really an optimised library, it would take far more resources to make it so, systematic profiling, hardware specific optimisation, etc.. something completely outside of our reach for the moment. And IMO, these small optimisations will not make a significant difference.
* On the other hand it has a real educational vocation, as a reference library of shader knowledge. And these kind of constructs distract from this goal, by reducing readability.

Again, my personal opinion. But at the end of the day you're in charge of which direction the project takes and I'll follow your lead on this.

Reply:
I will bring my [benchmarking tool back into shape](https://github.com/patriciogonzalezvivo/glslBenchmark). What if once the new feature are settle, the micro-optimizations are only for GLSL? I feel that having optimizations per-language is what make this very manual model worth it.

Reply:
This is a great point. I will think on it

Reply:
@shadielhajj I hear you. Per-language optimizations are a distraction and introduce friction. Specially for big re-factoring like you are doing right now. 
Also, is a reason to choose LYGIA over language translators. Is an ambitious goal, but hopefully one that keep us learning and improving. Keep us open to others to jump in to help and contribute. I prefer if we don't close the door to it. But also I don't want to slow you down. Let's not do micro optimizations at this stage. Once things settle down and we are not changing things all the time. When there is less need for agility. I can, if you are ok, came back to this and take on my self to apply them.

Reply:
Sounds good @patriciogonzalezvivo! Great to hear we're aligned on this. I would only add that when the time for optimisation comes, I recommend we do it using benchmarking/profiling. In my experience, the greatest bottlenecks are often not where one would expect them to be.

Reply:
agree. Thinking on doing per function (file) test on different hardware.  Doing some first prototypes here: https://github.com/patriciogonzalezvivo/lygia_examples